Indexed db extension

ffghdfh opened the pull request:
Added an IndexedDB extension.

CST1229 commented on extensions/ffghdfh/indexedDB.js:
```diff
@@ -0,0 +1,221 @@
+  
+        return new Promise((resolve, reject) => {
+          request.onsuccess = () => resolve(request.result ? request.result.value : null);
+          request.onerror = () => reject(request.error);
```
Slight issue, in unsandboxed extensions [blocks throwing errors can break stuff](https://docs.turbowarp.org/development/extensions/unsandboxed#increased-power-brings-increased-responsibility). You should probably just resolve with an empty string (for reporters) or undefined (for commands) in error cases. (This also applies to all other times `reject()` is used in blocks.)

ffghdfh replied:
Oh gee, didn't know that! Thanks for letting me know! I'm rally new to github though, so how do I modify my changes 'n stuff?

ffghdfh commented:
I removed all the reject()-s, replaced 'em all with resolve()-s, although i'm a bit tired, it's getting late, so it might be a bit lame of a solution 😅
